test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs

```diff
@@ -21,6 +21,17 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
         public HttpClient Client { get; }
 
         [Fact]
+        public async Task RazorPage_TempDataFilters_DontSurviveBetweenPages()
+        {
+            var response = await Client.GetAsync("http://localhost/TempData/TempDataPageModelProperty");
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            //If the filter from above was re-used in the below request the request would fail due to type cast.
+            response = await Client.GetAsync("http://localhost/HandlerTestPage");
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

Comment:
Or do we think that's the kind of bug that will only happen while writing it and we don't need the test?